server/src/main/java/org/opensearch/action/main/TransportMainAction.java

```diff
@@ -33,36 +33,51 @@
 package org.opensearch.action.main;
 
 import org.opensearch.Build;
+import org.opensearch.LegacyESVersion;
 import org.opensearch.Version;
 import org.opensearch.action.ActionListener;
 import org.opensearch.action.support.ActionFilters;
 import org.opensearch.action.support.HandledTransportAction;
 import org.opensearch.cluster.ClusterState;
 import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Setting;
 import org.opensearch.common.settings.Settings;
 import org.opensearch.node.Node;
 import org.opensearch.tasks.Task;
 import org.opensearch.transport.TransportService;
 
 public class TransportMainAction extends HandledTransportAction<MainRequest, MainResponse> {
-
     private final String nodeName;
     private final ClusterService clusterService;
+    private final String responseVersionNumber;
+
+    // HTTP_MAIN_RESPONSE_VERSION_NUMBER exists to provide support for clients expecting legacy ES versions
+    // to be returned from the main rest endpoint.  This setting will be removed in future versions of OpenSearch.
+    public static final Setting<String> SETTING_HTTP_MAIN_RESPONSE_VERSION_NUMBER =
```

Comment:
I still suggest using system property instead of node property in yml as this is an advance control and generally not recommended. It is a hack to unblock clients. Deprecating system property would be lot easier as compared to setting in yml and both will serve the same purpose.

Reply:
agree - will update this